samcli/lib/providers/sam_function_provider.py

```diff
@@ -177,12 +196,13 @@ def _extract_sam_function_codeuri(name, resource_properties, code_property_key):
         # CodeUri can be a dictionary of S3 Bucket/Key or a S3 URI, neither of which are supported
         if isinstance(codeuri, dict) or (isinstance(codeuri, str) and codeuri.startswith("s3://")):
             codeuri = SamFunctionProvider._DEFAULT_CODEURI
-            LOG.warning(
-                "Lambda function '%s' has specified S3 location for CodeUri which is unsupported. "
-                "Using default value of '%s' instead",
-                name,
-                codeuri,
-            )
+            if not ignore_code_extraction_warnings:
```

Comment:
Why are we changing this warning?

Reply:
since we pass in a packaged template from the deploy context where the code uri has been changed to an s3 uri. Without that ignore code extractions, it will throw a message saying code uri with s3 is not supported.